### PR TITLE
Vagrant cleanup

### PR DESCRIPTION
**Pull request by jgrund**
By using an internal named network and dynamic dns, we can drop dhcp and
parsing of dynamic ips.

We can also generate private / public keys when the box starts, and
remove them from SCM.

Signed-off-by: Joe Grund <joe.grund@intel.com>

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -1,50 +1,60 @@
...
+  SHELL
+
+  # Create device-scanner node
+  SCANNER_NAME = 'device-scanner'.freeze
+  config.vm.define "#{SCANNER_NAME}#{NAME_SUFFIX}", primary: true do |device_scanner|
+    device_scanner.vm.hostname = SCANNER_NAME
+    device_scanner.ssh.username = 'root'
+    device_scanner.ssh.password = 'vagrant'
+    device_scanner.vm.network :forwarded_port,
+                              host: 8080,
```
Comment (johnsonw) [inline Vagrantfile]: Thinking about this more, do we use 8080?

File: Vagrantfile [hunk 2]
```diff
@@ -1,50 +1,60 @@
...
+                              guest: 8080,
+                              auto_correct: true
+    device_scanner.vm.network 'private_network',
+                              ip: '10.0.0.10',
```
Comment (johnsonw) [inline Vagrantfile]: Are you against making the IP's constants at the top of the file? I think that might be a bit cleaner.

File: Vagrantfile [hunk 3]
```diff
@@ -55,61 +65,67 @@ Vagrant.configure("2") do |config|
...
         v.customize ['setextradata', :id, "VBoxInternal/Devices/ahci/0/Config/Port#{i}/SerialNumber", "081118FC1221NCJ6G8#{id}"]
       end
     end
 
-    device_scanner.vm.hostname = $device_scanner_hostname
-    device_scanner.vm.network "private_network", type: "dhcp"
-    device_scanner.vm.network :forwarded_port, host: 8080, guest: 8080, auto_correct: true
+    device_scanner.vm.provision 'deps', type: 'shell', inline: <<-SHELL
+      yum install -y http://download.zfsonlinux.org/epel/zfs-release.el7_5.noarch.rpm
+      # yum -y copr enable managerforlustre/manager-for-lustre-devel
```
Comment (johnsonw) [inline Vagrantfile]: Will we be able to re-enable this any time soon or should we just remove it?

File: Vagrantfile [hunk 4]
```diff
@@ -119,109 +135,56 @@ Vagrant.configure("2") do |config|
...
+      npm i --ignore-scripts
+      cert-sync /etc/pki/tls/certs/ca-bundle.crt
+      scl enable rh-dotnet20 "npm run restore"
+    SHELL
 
-cat >/root/.ssh/config<<__EOF
-Host testnode
-  Hostname $test_node_ip
-  StrictHostKeyChecking no
+    test.vm.provision 'testing', type: 'shell', inline: <<-SHELL
```
Comment (johnsonw) [inline Vagrantfile]: Should we name this `integration-test`?

File: test.sh [hunk 5]
```diff
@@ -1,4 +1,4 @@
-vagrant destroy -f
 export NAME_SUFFIX=$(pwd | sha256sum | head -c 32)
+vagrant destroy -f device-scanner$NAME_SUFFIX test$NAME_SUFFIX
 vagrant up device-scanner$NAME_SUFFIX test$NAME_SUFFIX
-vagrant destroy -f
+vagrant destroy -f device-scanner$NAME_SUFFIX test$NAME_SUFFIX
```
Comment (johnsonw) [inline test.sh]: Can't we do:
```
vagrant destroy -f
vagrant up
vagrant destroy -f
```

File: IML.IntegrationTestFramework/source/IntegrationTestFramework.fs [hunk 6]
```diff
@@ -22,7 +22,7 @@ type RollbackStateResult<'a, 'b> = Result<'a * RollbackState, 'b * RollbackState
 type CommandResponseResult = Result<string * string * string, string * string * string>
 
 let shellCommand: string -> string =
-  sprintf "ssh devicescannernode '%s'"
+  sprintf "ssh -o LogLevel=error 10.0.0.10 '%s'"
```
[the remark this answers is not in the document]
Comment (jgrund) [reply IML.IntegrationTestFramework/source/IntegrationTestFramework.fs]: I'm not sure there is an advantage to that over this approach.

File: IML.IntegrationTestFramework/test/IntegrationTest.fs [hunk 7]
```diff
@@ -123,7 +123,7 @@ testAsync "Stateful Promise should stop executing commands and rollback when an
               failwithf "Command result should have matched the error case."
             | Error (e, _) ->
               let myError = e |> errToString
-              myError == "{\"killed\":false,\"code\":127,\"signal\":null,\"cmd\":\"ssh devicescannernode 'ech \\\"goodbye\\\"'\"} - \"bash: ech: command not found\\n\""
+              myError == "{\"killed\":false,\"code\":127,\"signal\":null,\"cmd\":\"ssh -o LogLevel=error 10.0.0.10 'ech \\\"goodbye\\\"'\"} - \"bash: ech: command not found\\n\""
```
[the remark this answers is not in the document]
Comment (jgrund) [reply IML.IntegrationTestFramework/test/IntegrationTest.fs]: Really, we are just logging the call, why does an environment make this better?

File: Vagrantfile (hunk 1 above)
Comment (jgrund) [reply Vagrantfile]: We use it to debug remotely using the node debugger.

File: Vagrantfile (hunk 3 above)
Comment (jgrund) [reply Vagrantfile]: I tried re-enabling it with this patch, it's still broken.

This is the correct way to do it, so I'd like to leave it in until it works again

File: test.sh (hunk 5 above)
Comment (jgrund) [reply test.sh]: This is where there is still an issue with boxes colliding. If you kick off two runs once one of them finishes and destroys itself the other box will exit as well. 

I tried this as a way to namespace the destroy, but there is something that still overlaps.

File: Vagrantfile [hunk 8]
```diff
@@ -1,50 +1,60 @@
...
-   config.vm.provision "file", source: "id_rsa", destination: "/root/.ssh/id_rsa"
-   config.vm.provision "shell", inline: $setup_public_key
-   config.vm.provision "shell", inline: $set_key_permissions
-
-  #
-  # Create a device-scanner node
-  #
-  config.vm.define "device-scanner#{$suffix}", primary: true do |device_scanner|
-    device_scanner.vm.provider "virtualbox" do |v|
+NAME_SUFFIX = ENV['NAME_SUFFIX'] || ''
```
[the remark this answers is not in the document]
Comment (jgrund) [reply Vagrantfile]: sure

File: IML.IntegrationTestFramework/test/IntegrationTest.fs (hunk 7 above)
[the remark this answers is not in the document]
Comment (johnsonw) [reply IML.IntegrationTestFramework/test/IntegrationTest.fs]: I'm ok with it, just feels odd that we have "10.0.0.10" multiple times.

File: Vagrantfile (hunk 1 above)
Comment (johnsonw) [reply Vagrantfile]: ah, right 👍

File: Vagrantfile (hunk 3 above)
Comment (johnsonw) [reply Vagrantfile]: 👍

File: test.sh (hunk 5 above)
Comment (johnsonw) [reply test.sh]: Ok, guess we will have to keep an eye out.

File: IML.IntegrationTestFramework/test/IntegrationTest.fs (hunk 7 above)
[the remark this answers is not in the document]
Comment (jgrund) [reply IML.IntegrationTestFramework/test/IntegrationTest.fs]: Ah, you are talking about the ip. Given the network is static, I don't see much advantage to using a hostname or passing in an env, it's just a global find -> replace.

File: Vagrantfile (hunk 2 above)
Comment (jgrund) [reply Vagrantfile]: meh, this is only used once.

File: Vagrantfile (hunk 4 above)
Comment (jgrund) [reply Vagrantfile]: Sure

**Pull request merged**